src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -2884,6 +2884,206 @@ export default class EthereumApi implements Api {
+   * Added in [EIP-1559](https://www.alchemy.com//blog/eip-1559)
+   *
+   * @param blockCount - Range of blocks between 1 and 1024. Will return less than the requested range if not all blocks are available.
+   * @param newestBlock - Highest block of the requested range.
+   * @param rewardPercentiles - A monotonically increasing list of percentile values. For each block in the requested range,
+   * the transactions will be sorted in ascending order by effective tip per gas and the corresponding effective tip for the percentile
+   * will be determined, accounting for gas consumed.
+   * @returns transaction base fee per gas and effective priority fee per gas for the requested/supported block range
+   *
+   * * `oldestBlock`:  - Lowest number block of the returned range.
+   * * `baseFeePerGas`:  - An array of block base fees per gas. This includes the next block after the newest of the returned range,
+   * because this value can be derived from the newest block. Zeroes are returned for pre-EIP-1559 blocks.
+   * * `gasUsedRatio`:  - An array of block gas used ratios. These are calculated as the ratio of `gasUsed` and `gasLimit`.
+   * * `reward`:  - An array of effective priority fee per gas data points from a single block. All zeroes are returned if the
+   * block is empty.
+   *
+   * @example
+   * ```javascript
+   * const [from, to] = await provider.request({ method: "eth_accounts", params: [] });
+   * await provider.request({ method: "eth_sendTransaction", params: [{ from, to }] });
+   * const feeHistory = await provider.request({ method: "eth_feeHistory", params: ["0x1", "0x1", [10, 100]] });
+   * console.log(feeHistory);
+   * ```
+   */
+  @assertArgLength(3)
+  async eth_feeHistory(
+    blockCount: QUANTITY,
+    newestBlock: QUANTITY | Ethereum.Tag,
+    rewardPercentiles: number[]
+  ): Promise<Ethereum.FeeHistory<"private">> {
+    const blockchain = this.#blockchain;
+    const MIN_BLOCKS: number = 1;
+    const MAX_BLOCKS: number = 1024;
+    const PRECISION_FLOAT: number = 1e14;
+    const PAD_PRECISION = 16;
+    const PRECISION_BIG_INT: bigint = BigInt(1e16);
+
+    const newestBlockNumber = blockchain.blocks
+      .getEffectiveNumber(newestBlock)
+      .toBigInt();
+
+    // blockCount must be within MIN_BLOCKS and MAX_BLOCKS. blockCount > newestBlock is
+    // technically valid per the spec but we cannot go past the Genesis Block. Values
+    // above MAX_BLOCKS are technically within spec, however we cap totalBlocks because
+    // of the resource needs and potential abuse of a very large blockCount.
+    const totalBlocks = Math.min(
+      Math.max(
+        Quantity.toBigInt(blockCount) >= MAX_BLOCKS
+          ? MAX_BLOCKS
+          : Quantity.toNumber(blockCount),
+        MIN_BLOCKS
+      ),
+      Quantity.toNumber(
+        newestBlockNumber >= MAX_BLOCKS ? MAX_BLOCKS : newestBlockNumber + 1n
+      ),
+      MAX_BLOCKS
+    );
```

Comment:
:+1: https://github.com/trufflesuite/ganache/pull/3351/commits/6fbf0c5fbf538c94f8e0455ef1c679425556eb6a#diff-9597e9aea6d27393d0f6651003a17315b36cc3b86ccd5feb1d03905474f3fd55



Let me know what you think. Reduce will throw without defaults for the accumulator. Maybe types are enough for this? Trying to get rid of defensive prog urges because typescript.